overlord/snapstate/snapstate.go

```diff
@@ -641,6 +641,20 @@ func TryPath(st *state.State, name, path string, flags Flags) (*state.TaskSet, e
 //
 // The returned TaskSet will contain a DownloadAndChecksDoneEdge.
 func Install(st *state.State, name, channel string, revision snap.Revision, userID int, flags Flags) (*state.TaskSet, error) {
+	return installGeneric(st, name, channel, "", revision, userID, flags)
+}
+
+// InstallCohort is like Install but instead of a revision, it takes a cohort key.
+func InstallCohort(st *state.State, name, channel, cohortKey string, userID int, flags Flags) (*state.TaskSet, error) {
```

Comment:
is cohort something we install, we are installing the snap?  or do we install "from" it or some other preposition?

Reply:
I don't particularly mind either way: `InstallFromCohort` is more correct, but also longer, and I don't feel that the added information makes things better -- nor does the longer name make it worse.

Reply:
we agreed because of the refactoring needed later to keep like this anyway